blog/2022/08-30-discover-kubernetes/index.md

```diff
@@ -0,0 +1,303 @@
+---
+authors: [matthias]
+tags: [cloud, search, graph, k8s]
+image: ./img/banner-social.png
+---
+
+# Discover Kubernetes Resources
+
+Kubernetes has dramatically improved the way we manage our workloads. It has become the de facto standard for deploying and managing containers.
+
+A typical setup uses different Kubernetes clusters for different application stages (dev, test, prod) or one cluster per tenant. A Kubernetes cluster shared between different users and teams usually uses namespaces and roles to control access. Deploying a single application to a Kubernetes cluster usually consists of 10s to 100s of resources to deploy (deployments, services, configmaps, secrets, ingresses, etc.).
+
+What starts simple in the beginning becomes tedious quite fast since the number of resources is vast and the view of a single user is limited.
+
+![Kubernetes](./img/banner.png)
+
+<!--truncate-->
+
+## Resoto Overview
+
+Resoto can provide a consistent view of all resources in all Kubernetes clusters in all namespaces by continuously scraping the Kubernetes API of all configured clusters. The collected information can be accessed using [ResotoShell](/docs/concepts/components/shell), [ResotoNotebook](https://resoto.com/docs/reference/notebook) or ResotoUI.
+
+![Resoto Overview](img/resoto_overview.png)
+
+:::note The examples that I will present in this blog post are also available as a separate notebook [here](https://github.com/someengineering/resotonotebook/blob/main/examples/resoto_kubernetes.ipynb), so you can play with your Kubernetes cluster once you have installed Resoto. :::
+
+## Full-text search
+
+A simple and powerful way of finding resources in your cluster is by using the full-text search index that Resoto maintains. It indexes all properties of every service in every namespace in every cluster. Everything we need to do is enter the search term enclosed in double quotes.
+
+```shell
+> search "10.245.11.204"
+kind=kubernetes_service, id=377f0, name=test-service, age=6d23h, cloud=k8s, account=dev, region=test
+```
+
+This search for an IPv4 address returned one resource of kind `kubernetes_service`. To see all properties of this service, we need to pipe the search into the `dump` command. Here we can see the IPv4 address as the cluster IP we searched.
```

Comment:
we should link to where we cover the dump command in the docs: https://resoto.com/docs/reference/cli/dump